src/data_info/services/stat.clj

```diff
@@ -46,76 +65,107 @@
+        includes-set (get-filter-set include all-keys)
+        excludes-set (get-filter-set exclude [])]
+      (cset/intersection all-keys (cset/difference includes-set excludes-set))))
 
 (defn ^IPersistentMap path-stat
-  [^IPersistentMap cm ^String user ^String path]
-  (let [path (ft/rm-last-slash path)]
+  [^IPersistentMap cm ^String user ^String path & {:keys [filter-include filter-exclude] :or {filter-include nil filter-exclude nil}}]
+  (let [path (ft/rm-last-slash path)
+        included-keys (process-filters filter-include filter-exclude)]
     (log/debug "[path-stat] user:" user "path:" path)
     (validators/path-exists cm path)
-    (decorate-stat cm user (info/stat cm path))))
+    (let [base-stat (if (needs-any-key? included-keys :type :date-created :date-modified :file-size :md5)
+                      (info/stat cm path)
+                      {:path path})]
+      (decorate-stat cm user base-stat included-keys))))
 
 (defn ^IPersistentMap uuid-stat
-  [^IPersistentMap cm ^String user uuid]
+  [^IPersistentMap cm ^String user uuid & {:keys [filter-include filter-exclude] :or {filter-include nil filter-exclude nil}}]
   (log/debug "[uuid-stat] user:" user "uuid:" uuid)
   (let [path (uuid/get-path cm uuid)]
-    (path-stat cm user path)))
+    (path-stat cm user path :filter-include filter-include :filter-exclude filter-exclude)))
 
 (defn do-stat
-  [{user :user validation :validation-behavior} {paths :paths uuids :ids}]
+  [{:keys [user validation-behavior filter-include filter-exclude] :or {filter-include nil filter-exclude nil}} {paths :paths uuids :ids}]
   (irods/with-jargon-exceptions [cm]
     (validators/user-exists cm user)
     (validators/all-uuids-exist cm uuids)
     (let [uuid-paths (map (juxt (comp keyword str) (partial uuid/get-path cm)) uuids)
           all-paths (into paths (map second uuid-paths))]
       (validators/all-paths-exist cm all-paths)
-      (case (keyword validation)
+      (case (keyword validation-behavior)
             :own (validators/user-owns-paths cm user all-paths)
             :write (validators/all-paths-writeable cm user all-paths)
             :read (validators/all-paths-readable cm user all-paths)
             (validators/all-paths-readable cm user all-paths))
-      {:paths (into {} (map (juxt keyword (partial path-stat cm user)) paths))
-       :ids (into {} (map (juxt first #(path-stat cm user (second %))) uuid-paths))})))
+      {:paths (into {}
+                    (map
+                      (juxt keyword
+                            (fn [path]
+                              (path-stat cm user path :filter-include filter-include :filter-exclude filter-exclude)))
+                      paths))
+       :ids   (into {}
+                    (map
+                      (juxt first
+                            (fn [path-vec]
+                              (path-stat cm user (:second path-vec) :filter-include filter-include :filter-exclude filter-exclude)))
```

Comment:
ooh, probably it should. I'll double-check the endpoints still work when you pass IDs, but I suspect it does need to be changed